index.md

```diff
@@ -2,9 +2,286 @@
+If there is some issue with the internal registry that is fixed, it is likely that the user-facing version would need to be fixed as well – for a replica this is trivial but for a distinct registry this becomes an operational overhead.
+
+For this reason we must consider the option of a read-only user-facing registry containing the formal observatory data products, and a single user registry containing all the other datasets.
+As for the previous option, it will be necessary for the graph building to be modified to allow multiple registries (without which user batch processing is not possible) and some clarification as to how the user interacts with the two registries.
+
+One downside of a single user registry relates to the handling of collaborations.
+If datasets are written to a user run collection they can not currently be reallocated to a group run collection.
+Butler requires that the dataset be retrieved and then stored in the new location, resulting in the dataset UUID being changed and potentially breaking the provenance chain with the original dataset.
+
+### Writing Datasets
+
+Users will want to be able to store datasets in a butler repository.
+For Data Preview 0.2 {cite:p}`RTN-041` three buckets on Google were used: one for the raw data, one for the the data release products themselves, and one for user products.
+The first two buckets were read-only and the user bucket was accessed via the use of a chained datastore in the butler configuration.
+
+In operations a similar pattern will be used in that the raw data and formal data products must be stored in read-only locations, but with the additional constraint that users will have [quotas](#quotas) on the amount of storage they can use.
+
+There are many options where a dataset could be stored when the user issues `butler.put()`:
+
+1. Their personal POSIX storage space.
+2. Per-user and per-group buckets in the cloud or at the USDF.
+3. A general user bucket in the cloud or at the USDF per butler where `u/$USER/` is enforced.
+
+Option #1 allows a single storage area for all the user's data products and simplifies the calculation of quotas.
+It does not provide any means of supporting collaborative groups and that likely means this option is not tenable other than in the narrow "workspace" concept described in {cite:p}`DMTN-271`.
+
+The remaining options will all involve the server generating signed URLs for writes.
+
+Option #2 solves the collaboration problem at the expense of creating tens of thousands of buckets.
+Option #3 is similar to how we currently handle user data products but does require more work to handle quotas than option #2.
+
+The choice between USDF and Cloud storage buckets depends on the cost of cloud storage versus USDF storage (based on the expected size of user data products) and how user batch {cite:p}`DMTN-223` interacts with cloud users.
```

Comment:
Any time we're doing a tradeoff between USDF and Cloud storage I think we need to consider how that affects what image products we can cache in the Cloud, and until we can at least fit all of the coadds in Cloud storage I think that ought to be the priority.

Reply:
That reminds me I do not talk about the purported hybrid cache any where here. It doesn't really play well with the signed URLs that go directly to a bucket.